test/test_allocation_domain.cpp

```diff
@@ -1195,4 +1195,42 @@ TEST_F(NVFuserTest, AllocationDomainContiguityForExplicitBroadcast) {
   testValidate(fusion, outputs, {t0}, {t1}, __LINE__, __FILE__);
 }
 
+TEST_F(AllocationDomainTest, VectorizeOverlappingTensor) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(3);
+  fusion.addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = set(tv1);
+  fusion.addOutput(tv2);
+
+  // According to the stride order below, the allocation domain should be the
+  // same as the root domain. However, here we intentionally make the allocation
+  // domain [axis(1), axis(0), axis(2)] because doing so allows us to vectorize
+  // by 4.
+  tv0->setAllocationDomain(
+      {tv0->axis(1), tv0->axis(0), tv0->axis(2)}, {false, true, true});
+
+  for (auto tv : {tv2, tv1}) {
+    tv->reorder({{0, 1}});
```

Comment:
This shows my ignorance about nvFuser :), but can you clarify what this reorder is doing? I don't quite follow the comment there. 

```
// Reorder axes according to old2new[old_pos] = new_pos
```

Reply:
I added comment to before or after each step to the code:
```C++
  for (auto tv : {tv2, tv1}) {
    // [I0, I1, I2]
    tv->reorder({{0, 1}});
    // [I1, I0, I2]
    tv->merge(0);
    // [I1*I0, I2]
    tv->merge(0);
    // [I1*I0*I2]
    tv->split(0, 4);
    // [I1*I0*I2/4, 4]
    tv->axis(0)->parallelize(ParallelType::TIDx);
  }
```

Basically, how I think through the reorder is like this, for example, if I have a 5d tensor
```C++
[I0, I1, I2, I3, I4]
```
and I do `tv->reorder({{0, 1}, {1, 2}, {2, 3}})`, then I will think like this:

Let me start from
```
old: [I0, I1, I2, I3, I4]
new: [??, ??, ??, ??, ??]
```
Let me traverse the list one by one, the first item says "what was at 0 is moved to 1", then I will do:
```
old: [__, I1, I2, I3, I4]
new: [??, I0, ??, ??, ??]
```
Then the next item says "what was at 1 is moved to 2", which is:
```
old: [__, __, I2, I3, I4]
new: [??, I0, I1, ??, ??]
```
Then the next item says "what was at 2 is moved to 3", which is:
```
old: [__, __, __, I3, I4]
new: [??, I0, I1, I2, ??]
```
Now I have traversed all `old2new`, I will fill the remaining IterDomains from old to new in order:
```
old: [__, __, __, __, __]
new: [I3, I0, I1, I2, I4]
```

Reply:
> Now I have traversed all old2new, I will fill the remaining IterDomains from old to new in order:

Got it! That's what I missed.